app/uk/gov/hmrc/soletraderidentificationfrontend/models/SoleTraderDetails.scala

```diff
@@ -109,7 +157,31 @@ object SoleTraderDetails {
         case None => Json.obj()
       }
 
-    format.writes(soleTraderDetails) ++ formattedBusinessVerification ++ formattedIdentifiersMatch ++ formattedNinoInsights
+    val overseasTaxIdBlock: JsObject = (soleTraderDetails.optOverseasTaxIdentifier, soleTraderDetails.optOverseasTaxIdentifierCountry) match {
+      case (Some(overseasTaxId), Some(country)) => Json.obj(
+        "overseas" -> Json.obj(
+          "taxIdentifier" -> overseasTaxId,
+          "country" -> country
+        )
+      )
+      case (None, None) => Json.obj()
+      case _ => throw new InternalServerException("Error: Invalid combination of tax identifier and country")
+    }
+
+
+    format.writes(soleTraderDetails) ++ formattedBusinessVerification ++ formattedIdentifiersMatch ++ formattedNinoInsights ++ overseasTaxIdBlock
```

Comment:
A tricky one. Really writes should serialise an instance of SoleTraderDetails and as such that should include the overseas tax identifier values. However, here you want to replace them with the "overseas" block. I'm pretty certain from the database repository tests you can remove properties from a JsObject using <object>.-(Key name).

Reply:
Done